easybuild/easyblocks/t/tensorflow.py

```diff
@@ -0,0 +1,260 @@
+        else:
+            raise EasyBuildError("Failed to determine installation prefix for GCC")
+
+        inc_paths = [gcc_lib_inc, gcc_lib_inc_fixed, gcc_cplusplus_inc]
+        lib_paths = [gcc_lib64]
+
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            inc_paths.append(os.path.join(cuda_root, 'include'))
+            lib_paths.append(os.path.join(cuda_root, 'lib64'))
+
+        regex_subs = [
+            (r'-B/usr/bin/', '-B%s/ %s' % (binutils_bin, ' '.join('-L%s/' % p for p in lib_paths))),
+            (r'(cxx_builtin_include_directory:).*', '\n  '.join(r'\1 "%s"' % resolve_path(p) for p in inc_paths)),
+        ]
+        for tool in ['ar', 'cpp', 'dwp', 'gcc', 'gcov', 'ld', 'nm', 'objcopy', 'objdump', 'strip']:
+            path = which(tool)
+            if path:
+                regex_subs.append((os.path.join('/usr', 'bin', tool), path))
+            else:
+                raise EasyBuildError("Failed to determine path to '%s'", tool)
+
+        if self.toolchain.options.get('pic', None):
+            # -fPIE/-pie and -fPIC are not compatible, so patch out hardcoded occurences of -fPIE & -pie
+            regex_subs.extend([('-fPIE', '-fPIC'), ('"-pie"', '"-fPIC"')])
+
+        for path, dirnames, filenames in os.walk(self.cfg['start_dir']):
+            for filename in filenames:
+                if filename.startswith('CROSSTOOL'):
+                    full_path = os.path.join(path, filename)
+                    self.log.info("Patching %s", full_path)
+                    apply_regex_substitutions(full_path, regex_subs)
+
+        tmpdir = tempfile.mkdtemp(suffix='-bazel-build')
+
+        cmd = ['bazel', '--output_base=%s' % tmpdir, 'build']
+        # https://docs.bazel.build/versions/master/user-manual.html#flag--compilation_mode
+        cmd.append('--compilation_mode=opt')
+        # https://docs.bazel.build/versions/master/user-manual.html#flag--config
+        cmd.append('--config=opt')
+        # https://docs.bazel.build/versions/master/user-manual.html#flag--subcommands
+        # https://docs.bazel.build/versions/master/user-manual.html#flag--verbose_failures
+        cmd.extend(['--subcommands', '--verbose_failures'])
+
+        if self.toolchain.options.get('pic', None):
+            cmd.append('--copt="-fPIC"')
+
+        cmd.append(self.cfg['buildopts'])
+
+        # pass through environment variables that may specify location of license file for Intel compilers
+        for key in []:
+            if os.getenv(key):
+                cmd.append('--action_env=%s' % key)
+
+        if cuda_root:
+            cmd.append('--config=cuda')
+
+        if self.cfg['with_mkl_dnn']:
+            # this makes TensorFlow download & use mkl-dnn (cfr. https://github.com/01org/mkl-dnn)
+            # using the full Intel MKL doesn't work without additional effort...
+            cmd.extend(['--config=mkl'])
+
+        cmd.append('//tensorflow/tools/pip_package:build_pip_package')
```

Comment:
I am not sure what this does, but I going to guess that the double slash is a typo.

Reply:
No, it's not, that's how Bazel works... See https://www.tensorflow.org/install/install_sources#build_the_pip_package .